### PR TITLE
Reindex add-ons when authors username/display_name changes

### LINKED ISSUE #6964: Changing the developer account displayname is not reflected in listings.
I noticed when changing a displayname of an account it doesn't cause the display name of the add-ons to be updated to reflect the change. It's only when you carry out a save on the add-on that the name change update is reflected.

If the new user APIs will cause the relevant updates when writing changes then this issue can probably be closed.

STR:

* Look at an existing add-on listing page
* Go to the the user profile page and update your display name
* Go back to the listing and the add-on still shows the username from before the change.

### PR DESCRIPTION
**Pull request by diox**
Fix #6964

### CODE REVIEW

File: src/olympia/users/models.py [hunk 1]
```diff
@@ -489,13 +494,22 @@ class Meta:
...
     new_email, old_email = new_attr.get('email'), old_attr.get('email')
     if old_email and new_email != old_email:
         log.debug('Creating user history for user: %s' % instance.pk)
         UserHistory.objects.create(email=old_email, user_id=instance.pk)
+    # If username or display_name changes, reindex the user add-ons, if there
+    # are any.
+    if (new_attr.get('username') != old_attr.get('username') or
+            new_attr.get('display_name') != old_attr.get('display_name')):
+        from olympia.addons.tasks import index_addons
+        ids = [a.pk for a in instance.get_addons_listed()]
```
Comment (EnTeQuAk) [inline src/olympia/users/models.py]: one-character names are bad, even in list comprehensions. Let's call this `addon` maybe?

File: src/olympia/users/tests/test_models.py [hunk 2]
```diff
@@ -380,6 +380,53 @@ def test_empty_email(self):
...
 
 
+class TestOnChangeName(TestCase):
+    def setUp(self):
+        super(TestOnChangeName, self).setUp()
+
+        # We're in a regular TestCase class so index_addons should have been
+        # mocked.
+        from olympia.addons.tasks import index_addons
+        self.index_addons_mock = index_addons
```
Comment (EnTeQuAk) [inline src/olympia/users/tests/test_models.py]: I might be blind but where is this being set? I can only find it being used in src/olympia/addons/tests/test_commands.py

Comment (diox) [reply src/olympia/users/tests/test_models.py]: The mock? It comes from `start_es_mocks()`, which is called as a fixture in `conftest.py`.

Comment (EnTeQuAk) [reply src/olympia/users/tests/test_models.py]: eh, nevermind, I was simply blind earlier

**Pull request merged**